crates/core_simd/src/vector.rs

```diff
@@ -1090,6 +1090,72 @@ where
     }
 }
 
+macro_rules! impl_simd_array_conversions {
+    ($A:literal, $B:literal) => {
+        impl<T, U> From<Simd<T, { $A * $B }>> for [U; $B]
+        where
+            T: SimdElement,
+            Simd<T, $A>: Into<U>,
+        {
+            fn from(v: Simd<T, { $A * $B }>) -> Self {
+                // Safety: these have identical layout
```

Comment:
Is U constrained enough for this? Couldn't I make a ZST that implements From<Simd>?

Reply:
`U` is constrained enough, you can convert `Simd<u32, 16>` to
`[Box<Simd<u32, 4>>; 4]` if you like, which is fine because it indirectly invokes `Box::from` on each `Simd<u32, 4>`. it only requires the same layout for the first half of the conversion: `Simd<u32, 16>` to `[Simd<u32, 4>; 4]`

Reply:
if you have:
```rust
struct PrintFrom;
impl<T: std::fmt::Debug> From<T> for PrintFrom {
    fn from(v: T) -> Self {
        println!("{v:?}");
    }
}
```
you can convert `Simd<f32, 4>` to `[PrintFrom; 2]` and it'll print the two `Simd<f32, 2>` subvectors.

Reply:
whoops, I totally missed what was happening with the `map(Into::into)`